akvo/rsr/views/py_reports/kickstart_word_report.py

```diff
@@ -108,11 +116,231 @@ def render_report(request, project_id):
+
+    legend_table = doc.add_table(rows=4, cols=3)
+    legend_table.rows[0].cells[0].paragraphs[-1].add_run('Legend').bold = True
+
+    legend_table.rows[1].cells[0].paragraphs[-1].add_run().add_picture(load_image(IMG_GRADE_HIGH))
+    legend_table.rows[1].cells[1].text = '85% - 100%'
+    legend_table.rows[1].cells[2].text = 'Result nearly reached or reached'
+
+    legend_table.rows[2].cells[0].paragraphs[-1].add_run().add_picture(load_image(IMG_GRADE_MEDIUM))
+    legend_table.rows[2].cells[1].text = '50% - 84%'
+    legend_table.rows[2].cells[2].text = 'Result partly reached'
+
+    legend_table.rows[3].cells[0].paragraphs[-1].add_run().add_picture(load_image(IMG_GRADE_LOW))
+    legend_table.rows[3].cells[1].text = '0% - 49%'
+    legend_table.rows[3].cells[2].text = 'Result under reached'
+
+    doc.add_paragraph('')
+
+    quantitative_table = doc.add_table(rows=1, cols=3)
+    quantitative_table.cell(0, 0).merge(quantitative_table.cell(0, 2))
+    title_p = set_repeat_table_header(quantitative_table.rows[0]).cells[0].paragraphs[-1]
+    title_p.text = 'Results and indicators (quantitative)'
+    title_p.style = 'Heading 2'
+    for result in project_view.results:
+        if not result.has_quantitative_indicators:
+            continue
+        row = quantitative_table.add_row()
+        row.cells[0].merge(row.cells[2])
+        result_title_p = row.cells[0].paragraphs[-1]
+        result_title_p.text = result.title
+        result_title_p.style = 'Heading 3'
+        for indicator in result.indicators:
+            if not indicator.is_quantitative:
+                continue
+            row = quantitative_table.add_row()
+            row.cells[0].text = indicator.title
+            row.cells[1].text = indicator.progress_str
+            grade_image = IMG_GRADE_HIGH if indicator.grade == 'high' \
+                else IMG_GRADE_MEDIUM if indicator.grade == 'medium' \
+                else IMG_GRADE_LOW
+            row.cells[2].paragraphs[-1].add_run().add_picture(load_image(grade_image))
+
+    doc.add_paragraph('')
+
+    doc.add_heading('Results and indicators (qualitative)', 2)
+    for result in project_view.results:
+        if not result.has_qualitative_indicators:
+            continue
+        for indicator in result.indicators:
+            if not indicator.is_qualitative:
+                continue
+            doc.add_heading(indicator.title, 3)
+            for period in indicator.periods:
+                if not period.has_qualitative_data:
+                    continue
+                doc.add_paragraph('{} to {}'.format(
+                    period.period_start.strftime('%Y-%m-%d'), period.period_end.strftime('%Y-%m-%d')))
+                for update in period.approved_updates:
+                    markdown_to_docx(doc.add_paragraph(), update.narrative)
+                    if update.photo:
+                        doc.add_paragraph().add_run().add_picture(load_image(update.photo_url), width=Mm(100))
```

Comment:
Trying this on Project 2 crashed here. You could try it on Live data for project #2 and add some error handling here

Reply:
Should be fixed with the new commits.